### ISSUE TITLE
align `parse_grok` and `parse_groks` wrt handling field names

### ISSUE THREAD

**Issue by vladimir-dd** [vectordotdev/vector#10287]
`parse_grok`  and recently introduced `parse_groks`  functions have an important difference - `parse_groks`  treats dot-separated field names as lookup paths. We should align this behaviour. 

I suggest adding an optional parameter `support_nested_fields`(default: `false`) to the `parse_groks` function. This way we will align the default behaviour of both functions. 

Open question: 
- should we add the same parameter to `parse_grok`?

**Comment by JeanMertz** [comment]
@vladimir-dd could you give a few examples on when one or the other is useful? If we can resolve this without an extra function parameter `support_nested_fields`, that would be great. If there's a valid use-case to using both, then I agree we should allow one or the other "parse mode" to be used with a toggle such as the one your proposed.

**Comment by vladimir-dd** [comment]
@JeanMertz , treating field names as dot-separated paths is the current behaviour of DD grok parser, which is used a lot across Log Pipelines:
`%{data:nested.field}` -> `{ "nested": { "field": "value"}}`

**Comment by jszwedko** [comment]
I'd be inclined to make that the default behavior, but provide users a way to escape individual paths like we have for VRL lookups with quoting (e.g. `."foo.bar"`). Dd do you know if users can escape the `.` in Log Pipelines and what that looks like if so?

**Comment by vladimir-dd** [comment]
@jszwedko there is a way to escape dots in Log pipelines with `["..."]` like:
`%{data:abc[".xxx.yyy"]}`
but note that field names should still start with some parent field("abc"), so something like `%{data:["abc.xxx.yyy"]}` would not be allowed.

**Comment by jszwedko** [comment]
Right, yeah, that makes sense. I think I'd prefer to map Log Pipelines escaping into VRL's escaping for consistency within VRL. This would avoid needing to have people learn a new escaping syntax when working in VRL.

**Comment by vladimir-dd** [comment]
Then I guess we should escape dots with backslash according to [this](https://vector.dev/docs/reference/configuration/field-path-notation/#escaping)?

**Comment by JeanMertz** [comment]
> Then I guess we should escape dots with backslash according to [this](https://vector.dev/docs/reference/configuration/field-path-notation/#escaping)?

No, that's how escaping works within Vector itself, not VRL.

In VRL, escaping happens by using quotes, e.g. `.foo."bar.baz"`.

**Comment by jszwedko** [comment]
Yeah, what Jean said. We have an unfortunate difference between VRL escaping and template escaping (which is the docs you found). https://vector.dev/docs/reference/vrl/expressions/#path is the docs for VRL paths including a mention of escaping via quoting.

**Comment by uranru** [comment]
Did you manage to implement this?
I would like to update this request.
I am migrating from logstash, it works there.

**Comment by uranru** [comment]
Thanks for the comment.
I've been studying this product for a week now, comparing it with other solutions.
I already realized that grok is supported for formality. 
Apart from this problem, I didn't find how to process multiline logs through it.
That's why I'm testing regexp now.
